test/e2e/network_test.go

```diff
@@ -368,21 +368,31 @@ func getAffinityForNode(node *v1.Node) (*v1.Affinity, error) {
 	}, nil
 }
 
-// ensureNamespace checks if a namespace with the provided name exists and creates one if it does not
-func (tc *testContext) ensureNamespace(name string) error {
+// ensureNamespace checks if a namespace with the provided name exists and creates one if it does not with the given
+// labels
+func (tc *testContext) ensureNamespace(name string, labels map[string]string) error {
 	// Check if the namespace exists
-	_, err := tc.client.K8s.CoreV1().Namespaces().Get(context.TODO(), name, metav1.GetOptions{})
+	ns, err := tc.client.K8s.CoreV1().Namespaces().Get(context.TODO(), name, metav1.GetOptions{})
 	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
 	if err == nil {
+		// ensure namespace was properly created with the required labels
+		for k, expectedValue := range labels {
+			if foundValue, found := ns.Labels[k]; found && expectedValue != foundValue {
+				return fmt.Errorf("labels mismatch for namespace %s label: %s expected: %s found: %s",
+					name, k, expectedValue, foundValue)
+			}
+		}
+		// required labels present in namespace, nothing to do!
```

Comment:
I'm not sure this logic is complete. What if the namespace already exists but without a required label? We should by applying the missing ones

Reply:
>  What if the namespace already exists but without a required label?

Throws an error. The probability for this edge case to happen is very low since the namespace is maintained by the e2e test suite.

I was overthinking this, and now you are too 😃 .

Reply:
Fair enough

Reply:
The only use case I see is in our local testing where we are playing around with the labels and don't want the tests to fail if they don't exist.